ox_bqpipeline/bqpipeline.py

```diff
@@ -141,6 +142,66 @@ def wrapper(*args, **kwargs):
                         job.job_id)
     return wrapper
 
+def set_scalar_parameter(key, value, typ):
+    """Set a named scalar query parameter
+
+    :param key str: key for the named parameter
+    :param value (int, float, str, byte, datetime.datetime):
+        value for the named parameter
+    :param typ str: type of the scalar parameter.
+    :return: bigquery.ScalarQueryParameter
+    """
+    return bigquery.ScalarQueryParameter(key, bq_scalar_type_map[typ], value)
+
+def set_list_parameter(key, value, typ):
+    """Set a named array query parameter
+
+    :param key str: key for the named parameter
+    :param value list(int, float, str, byte, datetime.datetime):
+        value for the named parameter
+    :param typ str: type of the array parameter.
+    :return: bigquery.ArrayQueryParameter
+    """
+    typ = type(value[0]).__name__
```

Comment:
nit: better error handling and earlier type checking 
```suggestion
    if len(value) == 0:
        raise ValueError("BigQuery Array Parameter must be a non-empty list")
    if  len(set(map(type, value))) > 1:
        raise ValueError("All elements of BigQuery Array Parameter should be of the same type")
```

Reply:
Done.